src/PointwiseFunctions/GeneralRelativity/WeylElectric.cpp

```diff
@@ -0,0 +1,59 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/WeylElectric.hpp"
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace gr {
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ii<DataType, SpatialDim, Frame> weyl_electric(
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_ricci,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::II<DataType, SpatialDim, Frame>&
+        inverse_spatial_metric) noexcept {
```

Comment:
As long as this function is templated on DataType, I don't think it will compile if tensor_component is a double... I get the error `request for member ‘destructive_resize’ in ‘tensor_component’, which is of non-class type ‘double’`. Do you happen to have suggestions to get around this?

Reply:
Ah, yeah. I forgot that we didn't design that utility to be generic to the interior type of the tensor -- that will be a future feature for tensors. Instead of the full for-loop with the destructive resize, you should be able to:
```
if (UNLIKELY(get_size(get<0, 0>(*weyl_electric_part)) != get_size(get<0, 0>(inverse_spatial_metric)))) {
  *weyl_electric_part = tnsr::ii<DataType, SpatialDim, Frame>(get_size(get<0, 0>(inverse_spatial_metric)));
}
```
because the `get_size` function is designed with an overload for fundamental types, and the constructor for the tensor will work regardless of internal storage type, because it forwards its arguments to the construction of the internal storage type.